src/connector/ably-connector.ts

```diff
@@ -0,0 +1,128 @@
+import { Connector } from './connector';
+import { AblyChannel } from '../channel/ably-channel';
+import * as AblyImport from 'ably';
+import { PresenceChannel } from '../channel';
+import { AblyPresenceChannel } from '../channel/ably-presence-channel';
+
+/**
+ * This class creates a connector to Ably.
+ */
+export class AblyConnector extends Connector {
+    /**
+     * The Ably instance.
+     */
+    ably: AblyImport.Types.RealtimeCallbacks;
+
+    /**
+     * All of the subscribed channel names.
+     */
+    channels: Record<string, AblyChannel> = {};
+
+    /**
+     * Create a fresh Ably connection.
+     */
+    connect(): void {
+        if (typeof this.options.client !== 'undefined') {
+            this.ably = this.options.client;
+        } else {
+            this.ably = new Ably.Realtime(this.options);
+        }
+    }
+
+    /**
+     * Listen for an event on a channel instance.
+     */
+    listen(name: string, event: string, callback: Function): AblyChannel {
+        return this.channel(name).listen(event, callback);
+    }
+
+    /**
+     * Get a channel instance by name.
+     */
+    channel(name: string): AblyChannel {
+        if (!this.channels[name]) {
+            this.channels[name] = new AblyChannel(this.ably, name, this.options);
+        }
+
+        return this.channels[name];
+    }
+
+    /**
+     * Get a private channel instance by name.
+     */
+    privateChannel(name: string): AblyChannel {
+        // if (!this.channels['private-' + name]) {
+        //     this.channels['private-' + name] = new AblyPrivateChannel(this.ably, 'private-' + name, this.options);
+        // }
+        //
+        // return this.channels['private-' + name];
+        return this.channel(name);
```

Comment:
We need to follow ably namespaces https://ably.com/docs/general/channel-rules-namespaces in order to fit current echo interface properly. So, channels would be prefixed with 
1. Public channels -> public:channelName
2. Private channels -> private:channelName
3. Presence channels -> presence:channelName

Reply:
Ably pusher driver does implicit mapping for channels as per -> https://ably.com/tutorials/pusher-adapter#tutorial-step-4. Since, we are not using pusher driver, we need to define them explicitly. It will help us for assigning authz roles to users for specific type of channels.

Reply:
Resolving the conversation, since the code has been updated!